### PR TITLE
small fixes and IMessage

### PR DESCRIPTION
**Pull request by Vladiskos**

### CODE REVIEW

File: src/main/resources/config.properties [hunk 1]
```diff
@@ -1,9 +1,4 @@
-username=root
-<<<<<<< HEAD
-password=dimka1538
-url=jdbc:mysql://localhost/serverDb?allowPublicKeyRetrieval=true&useSSL=false&serverTimezone=UTC
-=======
+username=roo
```
Comment (ghost) [inline src/main/resources/config.properties]: root)

File: src/main/java/com/solvd/automation/lab/fall/listener/MyParser.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            if (request.matches(pattern.getPatternName())) {
+                return pattern.getParser();
+            }
+        }
+        return null;
+    }
+
+    public IMessage getResponse(String request) throws UnknownRequestType {
+        Parser parser = this.getParser(request);
+
```
Comment (shukal94) [inline src/main/java/com/solvd/automation/lab/fall/listener/MyParser.java:35]: Objects.requireNotNull(parser, "parser was not recognized")

File: src/main/java/com/solvd/automation/lab/fall/listener/MyParser.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        } else if (parser.getClass() == LogInParser.class) {
+            LogInMessage logInMessage = new LogInParser().parse(request);
+            return client.authenticate(logInMessage);
+        } else if (parser.getClass() == RegistrationParser.class) {
+            RegistrationMessage registrationMessage = new RegistrationParser().parse(request);
+            return client.registration(registrationMessage);
+        } else if (parser.getClass() == SearchParser.class) {
+            SearchMessage searchMessage = new SearchParser().parse(request);
+            return client.findClient(searchMessage);
+        } else if(parser.getClass() == ChecksumParser.class){
```
Comment (shukal94) [inline src/main/java/com/solvd/automation/lab/fall/listener/MyParser.java]: equals

File: src/main/java/com/solvd/automation/lab/fall/main/Server.java [hunk 4]
```diff
@@ -2,77 +2,35 @@
...
-                new Thread(()->{
-                    String request = socketConnector.readLine();
-                    Response response = Listener.getResponse(request);
-                    socketConnector.writeLine(response.toString());
-                    createSession("", currentClient, port);
-                }).start();
+                socketConnector = new SocketConnector(serverSocket);
+                ClientHandler clientHandler = new ClientHandler(socketConnector);
+                clientHandlers.add(clientHandler);
+                new Thread(clientHandler).start();
```
Comment (shukal94) [inline src/main/java/com/solvd/automation/lab/fall/main/Server.java]: add .join()

File: src/main/java/com/solvd/automation/lab/fall/model/ClientHandler.java [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+        return new ChecksumFromResponse(0, "checksum to "
+                + checksumMessage.getLoginOfRecipient() + " was sent");
+
+    }
+
+
+    public ChecksumFromResponse checksumTo(ChecksumMessage checksumMessage) {
+        for (ClientHandler c : Server.clientHandlers) {
+
```
Comment (shukal94) [inline src/main/java/com/solvd/automation/lab/fall/model/ClientHandler.java]: check if client token is not expired

File: src/main/java/com/solvd/automation/lab/fall/model/ClientHandler.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        } else {
+
+            return new SearchResponse(0, "" + searchingClient.getClientIp());
+        }
+    }
+
+    public ChecksumFromResponse checksumFrom(ChecksumMessage checksumMessage) {
+        ClientService clientService = new ClientService();
+
```
Comment (shukal94) [inline src/main/java/com/solvd/automation/lab/fall/model/ClientHandler.java]: see below

**Review by shukal94 — CHANGES_REQUESTED** [review]

File: logs/log_roll.txt [hunk 7]
```diff
@@ -310,6 +310,7 @@
 2020:-12-22 04:48:31 [Thread-1] INFO  com.solvd.automation.lab.fall.model.ClientHandler - Searching for Anna
 2020:-12-22 04:48:31 [Thread-1] INFO  com.solvd.automation.lab.fall.model.ClientHandler - Searched client: Client{clientIp='127.0.0.1:52906', clientToken=72781, login='Anna', passwordHash=2045663, sessionList=[]}
```
Comment (shukal94) [inline logs/log_roll.txt]: let's include that file to gitignore

**Review by shukal94 — CHANGES_REQUESTED** [review]
one minor comment

**Pull request merged**